### ISSUE TITLE
[BUG] [Ruby Client] Client-side cache uses a very generic global variable $data

### ISSUE THREAD

**Issue by jessiejuachon** [vmware/singleton#1457]
**Describe the bug**
Looking at some of the code for the cache it looks like it is using very generic terms as global variables ($data for instance.) This is really unsafe and could be overwritten by anyone. Check if class variables with @@data can be used.

**To Reproduce**
1. Use any client application that uses the VIP Ruby client library.
2. Add a global variable $data in the client application code ar runtime.
3. Notice that the global varible created in Step 2 overwrites the Singleton client-side cache.

**Expected behavior**
Singleton's client-side cache will not be overwritten.

**Screenshots**
If applicable, add screenshots to help you explain your issue

**Additional context**

**Comment by lyiyu66** [comment]
The issue is fixed with commit ecf8684fb9a7c634160ea25a6d3aec9e9b82c64f. The global variable $data in application will not overwrite the cache. From log file, it will send the http request one time as $data doesn't overwrite the previous cache.
```
@Result = SgtnClient::Translation.getString("JAVA", "helloworld", "fr")
puts "@Result: #@Result"
$data = nil
@Result = SgtnClient::Translation.getString("JAVA", "helloworld", "fr")
puts "@Result: #@Result"
```
But it introduces a new bug: https://github.com/vmware/singleton/issues/1465
Will verify it after new bug is fixed.

**Comment by lyiyu66** [comment]
The issue is fixed with commit 396c4ffdd7c23c287c206720b4e9e340029314e4.
